### PR TITLE
PIXAA-7: Leverage SpotRebalanceRecommendation for instance termination when available

### PR DESCRIPTION
**Pull request by JoelSpeed**
This updates the Azure spot termination handler implementation to understand a tech preview MS feature that is a SpotRebalanceRecommendation.

This new event type means you can get up to 15 minutes warning for nodes going away vs the 30 seconds you get with the preempt event. This should improve the likelihood of successful draining in the event of a spot instance going away

## Summary by CodeRabbit

* **New Features**
  * Treats spot rebalance recommendation events as valid termination signals alongside preempt events.

* **Improvements**
  * Detects and probes Azure scheduled-events API versions, automatically falling back between endpoints for more reliable polling.

* **Tests**
  * Reorganized and extended polling tests to cover both event types, verify stop/edge-case behavior, and ensure correct node-marking when notices occur.

### CODE REVIEW

File: pkg/termination/termination_test.go [hunk 1]
```diff
@@ -115,74 +114,141 @@ var _ = Describe("Handler Suite", func() {
...
+			Context("and the instance termination notice is not fulfilled", func() {
+				BeforeEach(func() {
+					httpHandler = newMockHTTPHandler(func(rw http.ResponseWriter, req *http.Request) {
+						atomic.AddInt32(&counter, 1)
+						emptyEvents(rw, req)
+					})
+				})
+
+				It("should not mark the node for deletion", func() {
+					Consistently(nodeMarkedForDeletion(testNode.Name)).Should(BeFalse())
```
Comment (nrb) [inline pkg/termination/termination_test.go:185]: Units are failing here. Stack trace says L464 is getting a nil pointer, so I think the logic for creating a node has changed with this PR.

File: pkg/termination/termination_test.go [hunk 2]
```diff
@@ -115,74 +114,141 @@ var _ = Describe("Handler Suite", func() {
 			close(stop)
 		})
 
-		It("should not return an error", func() {
+		PIt("should not return an error", func() {
 			Eventually(errs).Should(Receive(BeNil()))
```
Comment (nrb) [inline pkg/termination/termination_test.go:118]: Units panicking here.

File: pkg/termination/termination_test.go [hunk 3]
```diff
@@ -115,74 +114,141 @@ var _ = Describe("Handler Suite", func() {
...
+						// use 50ms since polling is set to 100ms
+						time.Sleep(50 * time.Millisecond)
+						continue
+					}
+
+					close(stop)
+				})
+
+				It("should not return an error", func() {
+					Eventually(errs).Should(Receive(BeNil()))
```
Comment (nrb) [inline pkg/termination/termination_test.go:218]: Units panicking here.

**Review by nrb — COMMENTED** [review]
Looks like the units are panicking in multiple places, all leading back to L464 of the `createNode` function.

**Comment by JoelSpeed** [comment]
/verified by @JoelSpeed 

I've tested this against the E2E that I've written up, we have prior verification of the main process here as well
